clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -858,6 +865,40 @@ class KernelObjVisitor {
   void VisitRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
                    const CXXRecordDecl *Wrapper, Handlers &... handlers);
 
+  // Base case, only calls these when filtered.
+  template <typename... FilteredHandlers, typename ParentTy>
+  void VisitUnion(const CXXRecordDecl *Owner, ParentTy &Parent,
+                  const CXXRecordDecl *Wrapper,
+                  FilteredHandlers &... handlers) {
+    (void)std::initializer_list<int>{
```

Comment:
See the change required earlier, the FilteredHandlers need to move to the front.  See my godbolt link earlier in this thread.

Reply:
Done